src/plugins/data/common/es_query/kuery/ast/ast.ts

```diff
@@ -17,21 +17,48 @@
  * under the License.
  */
 
-import _ from 'lodash';
+import { isUndefined } from 'lodash';
 import { nodeTypes } from '../node_types/index';
-import { parse as parseKuery } from './kuery';
-import { KQLSyntaxError } from '../errors';
+import { KQLSyntaxError } from '../kuery_syntax_error';
+import { KueryNode, JsonObject, DslQuery, KueryParseOptions } from '../types';
+import { IIndexPattern } from '../../../index_patterns/types';
 
-export function fromLiteralExpression(expression, parseOptions) {
+// @ts-ignore
+import { parse as parseKuery } from './_generated_/kuery';
+
+const fromExpression = (
+  expression: string | DslQuery,
+  parseOptions: Partial<KueryParseOptions> = {},
+  parse: Function = parseKuery
```

Comment:
nit: I wonder if `parse: Function` could be more strictly typed here.

like `parse: (expression: string | DslQuery, parseOptions: Partial<KueryParseOptions>) => Ast` or something along those lines